calibrate.py

```diff
@@ -342,11 +363,46 @@ def __init__(self):
                 self.board_config = json.load(fp)
                 self.board_config = self.board_config['board_config']
                 self.board_config_backup = self.board_config
+        else:
+            cameraProperties = self.device.getConnectedCameraFeatures()
+            calibData = self.device.readCalibration()
+            eeprom = calibData.getEepromData()
+            eeprom.productName = eeprom.productName.replace(" ", "-")
+            eeprom.boardName = eeprom.boardName.replace(" ", "-")
+            print(f"Product name: {eeprom.productName}, board name {eeprom.boardName}")
+            if eeprom.productName.split("-")[0] == "OAK":
+                detection = eeprom.productName.split("-")
+            elif eeprom.boardName.split("-")[0] == "OAK":
+                detection = eeprom.boardName.split("-")
+            else:
+                raise ValueError(f"Board config for Product name: {eeprom.productName}, board name {eeprom.boardName} not found.")
+            if "AF" in detection:
+                detection.remove("AF")
+            if "FF" in detection:
+                detection.remove("FF")
+            if "9782" in detection:
+                detection.remove("9782")
+            self.board_name = '-'.join(detection).upper()
+            board_path = Path(self.board_name)
+            if self.traceLevel == 1:
+                print(f"Board path specified as {board_path}")
+            if not board_path.exists():
+                board_path = (Path(__file__).parent / 'resources/depthai_boards/boards' / self.board_name.upper()).with_suffix('.json').resolve()
```

Comment:
This approach was discussed somewhere else as well. Cross apply

Reply:
Because some devices does not have name (when using the `device.getDeviceName` you get empty string) i will use try method as shown below:

```python
            try: 
                detection = self.device.getDeviceName()
                print(f"Device name: {detection}")
                detection = detection.split("-")
            except:
                cameraProperties = self.device.getConnectedCameraFeatures()
                calibData = self.device.readCalibration()
                eeprom = calibData.getEepromData()
                eeprom.productName = eeprom.productName.replace(" ", "-").upper()
                eeprom.boardName = eeprom.boardName.replace(" ", "-").upper()
                print(f"Product name: {eeprom.productName}, board name {eeprom.boardName}")
                if eeprom.productName.split("-")[0] == "OAK":
                    detection = eeprom.productName.split("-")
                elif eeprom.boardName.split("-")[0] == "OAK":
                    detection = eeprom.boardName.split("-")```
                    
Same argument. For some cameras hasAutofocus is not well defined in EEPROM or `dai.deviceInfo().`

Reply:
Mark down to be replaced with a getProductName & getDeviceName combo in v2.23 version of DepthAI
(aka if getProductName doesn't match anything, use getDeviceName)

Note, getDeviceName at the moment does what is written in the `try` case. Do you have a device where that is not the case?

Reply:
Ok, will do.

Yes, some old devices does not have DeviceName, but has productName or boardName. Thats why this is seperated.